### DISCUSSION TITLE
Help With Directory Error Read10X

### DISCUSSION THREAD

**Discussion by mannakade** [satijalab/seurat#5390] [Q&A]
I've come across this error before and entirely forgot how I resolved it- hoping someone here remembers!

Error message: 
Error in Read10X(data.dir = "C:/Users/*****/Desktop/CITE-seq data/skin 239/barcodes239.tsv.gz") : 
  Directory provided does not exist

Code for the message:
rna.data <- Read10X(data.dir = "C:/Users/*****/Desktop/CITE-seq data/skin 239/barcodes239.tsv.gz")

Here's all the libraries & packages I've tried loading in:
library(SeuratData)
library(ggplot2)
install.packages("Seurat")
library(Seurat)
install.packages(c('dplyr','patchwork'))
library(dplyr)
library(patchwork)

If I don't include the file name I get the error asking for a .tsv.gz file and I'm confident this should be the right line to load in a barcode file in order to make a ridge plot....

**Comment by samuel-marsh** [comment]
Hi,

The `data.dir` parameter is the path to folder containing the files it shouldn’t be path to single file.  Please paste the error you get when just using that path, as well as the names of files in that folder, and version of Seurat you are using.

Best,
Sam

**Reply by mannakade** [reply]
Thank you so much! Realized the mistake I was making again was specially naming the files instead of leaving them as "barcodes.tsv.gz" "features" "matrix" etc.

 `> rna.data <- Read10X(data.dir = "C:/Users/gabri/Desktop/CITE-seq data/skin 239")
Error in Read10X(data.dir = "C:/Users/gabri/Desktop/CITE-seq data/skin 239") : 
  Gene name or features file missing. Expecting features.tsv.gz
> rna.data <- Read10X(data.dir = "C:/Users/gabri/Desktop/CITE-seq data/skin 239")
10X data contains more than one type and is being returned as a list containing matrices of each type.
> rna <- CreateSeuratObject(counts = rna.data, project = "pbmc3k", min.cells = 3, min.features = 200)
Error in CreateAssayObject(counts = counts, min.cells = min.cells, min.features = min.features,  : 
  No cell names (colnames) names present in the input matrix
> rna <- CreateSeuratObject(counts = rna.data, project = "skin239", min.cells = 3, min.features = 200)
Error in CreateAssayObject(counts = counts, min.cells = min.cells, min.features = min.features,  : 
  No cell names (colnames) names present in the input matrix`

**Comment by samuel-marsh** [comment]
So the issue now is that it appears that your data contains more than just gene expression data.  That is why it returns a list instead of a matrix as the message details.  Therefore when using `CreateSeuratObject` you need to tell it which of the matrices in the list you want to use.  This can either be done using `rna.data[[#]]` to specify the matrix using numbers by order in the list that corresponds to the matrix of interest or by using the matrix name with backticks.

```
rna <- CreateSeuratObject(counts = rna.data[[1]], project = "skin239", min.cells = 3, min.features = 200)

# OR

rna <- CreateSeuratObject(counts = rna.data $`Gene Expression`, project = "skin239", min.cells = 3, min.features = 200)
```

Best,
Sam
